frontend/src/pages/commitments.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react"
+import { Link } from "gatsby"
+
+import Layout from "../components/layout"
+import Image from "../components/image"
+import SEO from "../components/seo"
+import Button from "react-bootstrap/Button"
+import ProgressBar from "react-bootstrap/ProgressBar"
+
+import Welcome from "../commitment_components/Welcome"
+import CommitmentsOverview from "../commitment_components/CommitmentsOverview"
+import Signup from "../commitment_components/Signup"
+
+const Commitments = () => {
+  const [step, setStep] = useState(0)
+  const [error, toggleError] = useState(false)
+
+  const showError = e => {
+    toggleError(true)
+  }
+
+  const prevStep = e => {
+    setStep(step - 1)
+    toggleError(false)
+  }
+  const nextStep = e => {
+    setStep(step + 1)
+    toggleError(false)
+  }
+
+  return (
+    <Layout>
+      <SEO title="Commitments" />
+
+      <Welcome stepVal={0} currentStep={step} onClick={nextStep} />
+      <CommitmentsOverview stepVal={1} currentStep={step} onClick={nextStep} />
+      <Signup stepVal={2} currentStep={step} onClick={nextStep} />
```

Comment:
This system of stepping between these components is clever and simple! I like it a lot.